### PR TITLE
SRCH-4003 bug: invalid thumbnail_url should not block indexing page c…

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Set value to `nil` for invalid `thumbnail_url`, so that it will not block indexing URL.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by MothOnMars — CHANGES_REQUESTED** [review]
@krbhavith , this looks like quite a bit more code than we need. I was thinking we could fix this in the `HtmlDocument#thumbnail_url` method:
```ruby
  def thumbnail_url
    url = metadata['og:image']&.first
    url&.match?(URI::DEFAULT_PARSER.make_regexp) ? url : nil
  end
```
Per our discussion with Amani, that code ensures `thumbnail_url` will return full, valid URLs and return `nil` otherwise.

**Comment by krbhavith** [comment]
I have added code to build absolute url before, now I reverted that and updated as per your suggesstion.

File: spec/models/html_document_spec.rb [hunk 1]
```diff
@@ -213,6 +213,20 @@
 
       it { is_expected.to eq 'http://www.foo.gov/og_image.jpg' }
     end
+
+    context 'when an og:image is not available' do
```
Comment (MothOnMars) [inline spec/models/html_document_spec.rb:217]: Please reword this to clarify what is being tested:
```ruby
context 'when og:image is not a valid URL' do
```

There should also be a spec to test that we return `nil` when there is no `og:image`.

**Review by MothOnMars — CHANGES_REQUESTED** [review]

File: spec/models/html_document_spec.rb [hunk 2]
```diff
@@ -213,6 +213,34 @@
 
       it { is_expected.to eq 'http://www.foo.gov/og_image.jpg' }
     end
+
+    context 'when og:image is not available' do
+      let(:raw_document) do
+        <<~HTML
+          <html lang="en">
+            <head>
+              <meta name="og:image" content="">
```
Comment (MothOnMars) [inline spec/models/html_document_spec.rb:222]: @krbhavith , I was actually hoping for a spec where the `og:image` meta tag isn't present at all. That possibility is the reason for the safe navigation operator, because without the `og:image` tag, this would fail:
```ruby
url.match?(URI::DEFAULT_PARSER.make_regexp)
```
while this would not:
```ruby
url&.match?(URI::DEFAULT_PARSER.make_regexp)
```

**Review by MothOnMars — APPROVED** [review]
